src/DocumentFormat.OpenXml/OpenXmlCompositeElement.cs

```diff
@@ -228,9 +228,9 @@ public bool AddChild(OpenXmlElement newChild, bool throwOnError = true)
         }
 
         /// <inheritdoc/>
-        [return: MaybeNull]
         [return: NotNullIfNotNull("newChild")]
-        public override T AppendChild<T>([AllowNull] T newChild)
+        public override T? AppendChild<T>(T? newChild)
+            where T : class
```

Comment:
This is a breaking change - we can't add a generic constraint where there wasn't already

Reply:
I understand that it looks like a breaking change, but in fact it is not. The generic constraint of the base method is inherited, and the base restricts T to be of type OpenXmlElement. 
Since OpenXmlElement is a class, this new constraint should not matter.

I am however not sure why we have to add the constraint, but it is the only way I could make it compile.
As I said, T is already restricted to OpenXmlElement, so why would you have to specify this constraint? I can try to research more.

Reply:
More information can be found here: https://devblogs.microsoft.com/dotnet/try-out-nullable-reference-types/, see "The issue with T?".

Reply:
Thanks for the clarification!I'm on vacation but will look when I get back